### ISSUE TITLE
Psalm crashes with InvalidArgumentException when it sees `@param T(): void $x`

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#978]
https://getpsalm.org/r/cc64d9ca97

Observed: Psalm crashes
Expected: Psalm does not crash; psalm optionally reports unextractable annotation on or around the offending line

This was seen when using `@param Closure():T` in a namespace (without namespace use or `\`), which is a syntax psalm deliberately  doesn't support

```
<?php

/** @param T():void $x */
function test($x) {
}
```

```
Scanning files...

Fatal error: Uncaught InvalidArgumentException: Unrecognised parse tree type Psalm\Type\ParseTree\MethodWithReturnTypeTree in /path/to/psalm/src/Psalm/Type.php:410
Stack trace:
#0 /path/to/psalm/src/Psalm/Type.php(119): Psalm\Type::getTypeFromTree(Object(Psalm\Type\ParseTree\MethodWithReturnTypeTree), false, Array)
#1 /path/to/psalm/src/Psalm/Visitor/DependencyFinderVisitor.php(1534): Psalm\Type::parseTokens(Array, false, Array)
#2 /path/to/psalm/src/Psalm/Visitor/DependencyFinderVisitor.php(1379): Psalm\Visitor\DependencyFinderVisitor->improveParamsFromDocblock(Object(Psalm\Storage\FunctionLikeStorage), Array, Object(PhpParser\Node\Stmt\Function_))
#3 /path/to/psalm/src/Psalm/Visitor/DependencyFinderVisitor.php(235): Psalm\Visitor\DependencyFinderVisitor->registerFunctionLike(Object(PhpParser\Node\Stmt\Function_))
#4 /path/to/psalm/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(181): Psalm\Visitor\DependencyFinderVisitor->e in /path/to/psalm/src/Psalm/Type.php on line 410
```

**Comment by weirdan** [comment]
Also see #974 (same underlying issue, I suspect)

**Comment by TysonAndre** [comment]
Yeah, it's a duplicate.

**Comment by muglug** [comment]
Fixed this specific issue in 3e920b1
